cart/application/cartService.go

```diff
@@ -1200,3 +1200,59 @@ func (qar QtyAdjustmentResults) HasRemovedCouponCodes() bool {
 
 	return false
 }
+
+// UpdateAdditionalData of cart
+func (cs *CartService) UpdateAdditionalData(ctx context.Context, session *web.Session, key string, value string) (*cartDomain.Cart, error) {
```

Comment:
Would expect here to pass in all additional data and not just a single key, guess the caller could do the merging/updating of a single one.